File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/models/GroupMember.java
package com.messaging.api.models;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@Document(collection = "groupMembers")
public class GroupMember extends BaseEntity {
    private String groupId;
    
    private String userId;
    
    private GroupRole role = GroupRole.Member;
    
    private LocalDateTime joinedAt = LocalDateTime.now();
}

File: DotNetMessaging/backend/DotNetMessaging.API/Models/GroupMember.cs
using MongoDB.Bson.Serialization.Attributes;

namespace DotNetMessaging.API.Models;

public class GroupMember : BaseEntity
{
    [BsonElement("groupId")]
    public string GroupId { get; set; } = string.Empty;

    [BsonElement("userId")]
    public string UserId { get; set; } = string.Empty;

    [BsonElement("role")]
    [BsonRepresentation(MongoDB.Bson.BsonType.String)]
    public GroupRole Role { get; set; } = GroupRole.Member;

    [BsonElement("joinedAt")]
    public DateTime JoinedAt { get; set; } = DateTime.UtcNow;
}

public enum GroupRole
{
    Member,
    Admin
}
